### PR TITLE
fix: implement methods with path-dependent type arguments

### PR DESCRIPTION
**Pull request by susliko**
Fixes https://github.com/scalameta/metals/issues/4668

### CODE REVIEW

File: mtags/src/main/scala-3/scala/meta/internal/pc/PcCollector.scala [hunk 1]
```diff
@@ -176,9 +176,11 @@ abstract class PcCollector[T](driver: InteractiveDriver, params: OffsetParams):
          */
         def isForComprehensionOwner(named: NameTree) =
           soughtNames(named.name) &&
-            named.symbol.owner.isAnonymousFunction && owners.exists(
-              _.span.point == named.symbol.owner.span.point
-            )
+            scala.util
```
Comment (tgodzik) [inline mtags/src/main/scala-3/scala/meta/internal/pc/PcCollector.scala:179]: Is this change needed? Does it solve any other issues?

**Review by tgodzik — APPROVED** [review]
Looks good! Just one question.

File: mtags/src/main/scala-3/scala/meta/internal/pc/PcCollector.scala (hunk 1 above)
Comment (susliko) [reply mtags/src/main/scala-3/scala/meta/internal/pc/PcCollector.scala:179]: Well, the fix works without this change, but some exception still pops out

Comment (tgodzik) [reply mtags/src/main/scala-3/scala/meta/internal/pc/PcCollector.scala:179]: ok sure, it make sense to add try there, thanks for the explanation!
